easybuild/easyblocks/j/java.py

```diff
@@ -27,13 +27,32 @@
 
 @author: Jens Timmerman (Ghent University)
 """
+from distutils.version import LooseVersion
 from easybuild.easyblocks.generic.packedbinary import PackedBinary
+from easybuild.tools.run import run_cmd
 
 
 class EB_Java(PackedBinary):
+#class PackedBinary(Binary, EasyBlock):
     """Support for installing Java as a packed binary file (.tar.gz)
     Use the PackedBinary easyblock and set some extra paths.
     """
+    
+    def extract_step(self):
+        """Unpack the source"""
+        #print LooseVersion(self.version)
+        #print self.src
+        print self.builddir
+        if LooseVersion(self.version) < LooseVersion('1.7'):
+            # uncompress java6 to build dir
+            cmd = 'cd %s; yes | bash %s' % (self.builddir, self.src[0]['path'])
```

Comment:
this is what you get...you just need to press enter that's why I did the `yes` trick...seems simpler to me but I will implement it as you prefer...

```
Java(TM) SE Development Kit 6 successfully installed.

Product Registration is FREE and includes many benefits:
* Notification of new versions, patches, and updates
* Special offers on Oracle products, services and training
* Access to early releases and documentation

Product and system data will be collected. If your configuration
supports a browser, the JDK Product Registration form will
be presented. If you do not register, none of this information
will be saved. You may also register your JDK later by
opening the register.html file (located in the JDK installation
directory) in a browser.

For more information on what data Registration collects and 
how it is managed and used, see:
http://java.sun.com/javase/registration/JDKRegistrationPrivacy.html

Press Enter to continue.....

 
Done.
```